### ISSUE TITLE
Possible logic problem with Travel Module

### ISSUE THREAD

**Issue by Vegeir** [MilleXIV/XIV_Databar#43]
Using 2.03 as 100 Fire Mage.

Right clicking on the travel options will not allow me to change anything.  I have my hearthstone, garrison hearthstone and Dalaran hearthstone with me.  Right and left click on Hearth (right) option seems to work properly.  The left option shows "Dalaran" and shows "teleport" as a cast bar when clicked.  What happens is that it uses my Dalaran hearthstone.  I am not high enough level to teleport to the new Dalaran so I am not sure if this is working as you intended.

**Comment by MilleXIV** [comment]
Did you get up to the point of getting your Order Hall port yet? Teleport: Hall of the Guardian? I have the same issue with my 100 Mage who hasn't done anything in Legion yet. I'm hoping there's some consistency there, which makes it easier to fix.

**Comment by Vegeir** [comment]
Yes, I had opened the Class Order Hall for the mage (and obviously, earning the spell) prior to trying this.  I just checked again.  Mage is now 101 and was sitting in the CoH.  I see the tooltip showing I can change my port options, but as has been said by others, I am unable to change my selection with a right-click.  The left-click on the "Dalaran" option on the left gives me a cast bar saying, "Teleport" and then burns my Dalaran Hearthstone CD.

![mage_port_options](https://cloud.githubusercontent.com/assets/21963210/18233153/610d5fd4-72ad-11e6-96b0-7e6d3538acb4.png)

**Comment by Vegeir** [comment]
Just checked Demon Hunter (101), Rogue (101) and Warrior (101).  They all have opened their CoH, have Hearthstones for Dalaran, Original and Garrison in their bags.  None are able to right click to change the option, however the tooltips are all visible:

![travel_options_rogue](https://cloud.githubusercontent.com/assets/21963210/18233226/fc7b90b6-72ae-11e6-90b5-37c14861ab10.png)

When I left-click on the word "Dalaran", this is what shows:

![travel_options_tpbar](https://cloud.githubusercontent.com/assets/21963210/18233236/60f9fb2c-72af-11e6-9360-bef8457ee413.png)

The icon is the Admiral's Compass (for Shipyard teleport), so I wasn't sure if you left it generic as opposed to using the hearthstone / name for the appropriate spell.  Just providing this info in case it helps.  Not being critical in any way.

**Comment by MilleXIV** [comment]
Okay, so I believe this is caused by having saved the selected option in the profile. Which was a bad idea and I shouldn't have done it. I've done a push that switches it to the char DB instead (which makes way more sense anyway).

The icon and spell name is not me. That's just how it is in WoW, nothing I can do about it.

**Comment by joev** [comment]
Having the same problem. No errors showing up just that the right-click doesn't work. Vegeir's screenshots are exactly like my experience. 

Any logs or anything we can supply to help you track this down @MilleXIV?

**Comment by joev** [comment]
I've pulled that update @MilleXIV but unfortuntately no joy. Issue remains unchanged.

**Comment by MilleXIV** [comment]
@joev Could you post your saved variables, either as an attachment here or as a [Gist](https://gist.github.com/).

**Comment by joev** [comment]
https://gist.github.com/joev/75680a96aa10dedcc58fcc39dba22c68

**Comment by MilleXIV** [comment]
Okay, wow, I just had an "well I'm dumb" moment. So it's definitely showing up. It just...happens to be showing up above the bar always. Should be fixed in 48a9ba3a409e96d5f8a658d315a533f93864fba7.

**Comment by joev** [comment]
@MilleXIV I admit... when I looked at my saved variables it did occur to me "Hmm... I wonder if it's cos my bar is at the top..." :) 

I'll pull that commit and let you know in about 30 seconds :)

**Comment by joev** [comment]
Works! Bug sorted. Glad it was as easy as positioning! :) Thanks @MilleXIV

**Comment by Amanthuul** [comment]
@Vegeir about spell name and icon - it's defaults for new dalaran hearthstone.

**Comment by Vegeir** [comment]
@Amanthuul  Yes, I didn't even realize until after @MilleXIV posted that and I went to check myself.  I wrote a WeakAura to track hearth CD and I am always matching (sometimes manually) the correct icons to my auras.  I had the Dalaran purple hearthstone graphic and thought it was the one used by default, which we know it isn't.  Can you tell who just pushed to 110 and didn't stop to smell the roses? :laughing:

**Comment by MilleXIV** [comment]
@Vegeir Not too surprising really depending on the class. With leveling my DK I only very rarely used it since I could just Death Gate and grab a portal from there.

Anyway, leaving this open because I had some weirdness happen to me yesterday where the list shrunk down to like 1 item when it should have had 3. Need to check if there's an event that fires when changing zones (I'm sure there is) and rebuild the list in that. Also, just realized I need to add an else around line 336.

**Comment by drveoj** [comment]
@MilleXIV I don't know if I should put this as an enhancement request or if it's related but I'd love if I could disable the tooltip (or have it move) so that when the dropdown (or popup) menu comes up it's not hidded by the tooltip.

**Comment by Vegeir** [comment]
@MilleXIV Try these:
1. **ZONE_CHANGED** (fires when the player moves between subzones or other named areas)
2. **ZONE_CHANGED_INDOORS** (fires when the player moves between areas and the "indoors/outdoors" status may have changed)
3. **ZONE_CHANGED_NEW_AREA** (fires when the player moves between major zones or enters/exits an instance)

**Comment by Amanthuul** [comment]
While you dead or in ghost, right click causes disappearance of hearthstones from list.
For shaman, changeable part totally disappear. For druid, only dreamwalk stays there.

**Comment by MilleXIV** [comment]
So I'm like 99% sure this has to do with IsUsableItem. Which seems to test if you can use the item at this exact moment instead of just in general. Going to switch it all to GetItemCount(id, false) instead. That will check if the item is in your bags.
